src/transformers/models/llava_next/image_processing_llava_next.py

```diff
@@ -464,6 +464,7 @@ def preprocess(
         return_tensors: Optional[Union[str, TensorType]] = None,
         data_format: Optional[ChannelDimension] = ChannelDimension.FIRST,
         input_data_format: Optional[Union[str, ChannelDimension]] = None,
+        concat_images: Optional[bool] = True,
```

Comment:
Rather than adding this flag to the image processor, we can do this inside the model right?

Just `pixel_values = pixel_values.view(-1, pixel_values.shape[2:])`

Reply:
the flag is not needed and can be removed, but we still need to concat the pixel_values at 
https://github.com/huggingface/transformers/blob/9b84e7dc3d2118c7f9f865ee63eb2300f897012c/src/transformers/models/llava_next/image_processing_llava_next.py#L610

If we have 2 images with 5 and 3 patches, then pixel_values is (8, 3, 336, 336), otherwise, it will cause error when converting to tensor.

Padding 3 patches into 5 patches and make it (2, 5, 3, 336, 336) requires you to undo the padding and stack it back to (8, 3, 336, 336) during model.forward

Reply:
Ok thanks for clarifying. Indeed we need PyTorch tensors as input to models in the Transformers library, we can't accept lists as the original implementation, so we need to stack them.

Alternatively, is there a possibility of padding to the max number of patches in a batch?

Reply:
Padding and then unpadding is possible if all inputs are required to have batch dimension, and can change to that if needed.

The current implementation is by "concat" because it would support multi-image in the same sequence (in-context learning).

If you dont mind, let me push a commit to support both padded-stack and concatenation (concat_images flag still needed but will default to False)

Reply:
@NielsRogge But are you able to reproduce the test?

Reply:
I'll try now from your branch